packages/compass-global-writes/src/store/index.spec.ts

```diff
@@ -89,6 +95,9 @@ function createStore({
       hasShardKey?: () => boolean | ShardKey;
       failsOnShardingRequest?: never;
       failsOnShardZoneRequest?: () => never;
+      failsToFetchClusterDetails?: never;
+      failsToFetchDeploymentStatus?: never;
+      failsToFetchShardKey?: () => boolean;
```

Comment:
This special test api being a catch-all for all the tests in this file is getting a bit hard to read from my perspective, especially because it seems like most of these methods are added for just one test case. If you need a one-off mocking / stubbing for a special test case like that I would really recommend doing it right in your test:

```
it('fails when ...', function() {
  sinon.stub(atlasService, 'authenticatedFetch').rejects(...)
  // ... more of the test here
})
```

That way you're not binding your test to a shared mock implementation which usually makes tests easier to maintain in the long run and easier to understand what's being mocked when reading through the test case

Reply:
I get your point, but in this case the tests were getting very verbose and difficult to read because 'authenticatedFetch' handles at 4 different things. So often the tests were repeating the whole [auth fetch mock logic](https://github.com/mongodb-js/compass/blob/fac4f9d038586f6f58d70fd0b04d5f37f158a986/packages/compass-global-writes/src/store/index.spec.ts#L96-L125 ) section only to change 1 thing. The only one that would be easy to mock and only used once would be `failsToFetchClusterDetails` - because that is the very first request that happens in the flow. That one I did just for consistency.